packages/runtime/container-runtime/src/gc/gcEarlyAdoption.md

```diff
@@ -0,0 +1,130 @@
+In other words, until you start using this, Tombstone enforcement will apply to all documents.
+
+### Advanced "Back door": Recovering and reviving Tombstoned objects
+
+If your application has Tombstone enabled and your users are encountering Tombstones - even at the point where
+Tombstone enforcement is enabled - there is a way to still access these objects to recover them and property
+reference them ("revival"). However, please understand that this is an advanced and unsupported path that may
+be immediately deprecated at any time.
+
+As mentioned above, bumping the gcTombstoneGeneration will free up impacted documents, but that's a permanent
+mitigation - those documents will never be exposed to GC Tombstone or Sweep.
+
+If there's a particular codepath in your application where objects being loaded may be Tombstoned,
+you may use this advanced "back door" to recover them and then properly reference them, thus restoring the document.
+For this to work, you must disable the `Fluid.GarbageCollection.ThrowOnTombstoneUsage` Config Setting.
+
+When a Tombstoned object (via `handle.get()`) fails to load, the 404 response error object has an `underlyingResponseHeaders` with the
+`isTombstoned` flag set to true: i.e. `error.underlyingResponseHeaders?.isTombstoned === true`. In this case,
+you may turn around and use `IContainerRuntime.resolveHandle` with `allowTombstone: true` in `IRequest.headers` to request
+the object again - this time it will succeed.
+
+To be very clear once again - This path uses deprecated APIs (`resolveHandle`) and comes with no guarantees of support.
+
+### Tombstones and the Summarizer Client
+
+Note: The Summarizer client will _never_ throw on usage or load of a Tombstoned object.
+
+## Enabling Sweep
+
+The following configuration is required for Sweep to be enabled for a given document:
+
+-   GC Option `gcSweepGeneration` must be set, and the persisted value must match the current value in the code
+-   Each of these two Config Settings must be set to `true` in the session:
+    -   `Fluid.GarbageCollection.Test.SweepDataStores`
+    -   `Fluid.GarbageCollection.Test.SweepAttachmentBlobs`
+
+### Differences between gcSweepGeneration and gcTombstoneGeneration
+
+`gcSweepGeneration` is persisted and immutable in the document, just like `gcTombstoneGeneration`.
+However, behavior differs in a few important ways.
+
+For Tombstone, if `gcTombstoneGeneration` is not set, Tombstone enforcement will be **enabled**.
+For Sweep however, if `gcSweepGeneration` is not set, Tombstone enforcement will be **disabled**.
+
+This means that until the `gcSweepGeneration` GC Option is set, _no existing document will be eligible for Sweep, ever_.
+So all documents created since the most recent bump to the gcSweepGeneration will have Sweep enabled.
+Note that if `gcSweepGeneration` is set and matches, Tombstone Mode is off for the session and `gcTombstoneGeneration` is ignored.
```

Comment:
Basically only one of them can be enabled at a time?

Reply:
Yeah if you're including `gcSweepGeneration` there's no reason to also include `gcTombstoneGeneration`. It will be persisted for new documents, but otherwise ignored (but in a future session with Sweep turned off it would matter).

Feel free to suggest alternate wording to make it clearer